variant_qc/exomes_genomes_coverage.py

```diff
@@ -0,0 +1,145 @@
+    gtf = gtf.annotate(locus=hl.range(gtf.interval.start.position, gtf.interval.end.position).map(lambda x: hl.locus(gtf.interval.start.contig, x, 'GRCh37')))
+    gtf = gtf.key_by().select('gene_id', 'locus', 'gene_name').explode('locus')
+    gtf = gtf.key_by('locus', 'gene_id').distinct()
+    return gtf.checkpoint('gs://gnomad-tmp/gencode_grch37.gene_by_base.ht', overwrite=True)
+
+def compute_per_base_cds_coverage():
+    """
+    Creates a HT with a row for each base and gene that is in a CDS in gencode v19 with the following information:
+    1) gene id
+    2) gnomAD exomes coverage by platform
+    3) gnomAD genomes coverage by PCR status
+    """
+
+    def get_haploid_coverage_struct(mt):
+        return hl.cond(
+            mt.locus.in_autosome_or_par(),
+            hl.struct(
+                mean=0.5*hl.agg.sum(mt.mean  * mt.n)/hl.agg.sum(mt.n),
+                median=0.5*hl.median(hl.agg.collect(mt.median)),
+                **{
+                    f'over_{i/2}': hl.agg.sum(mt[f'over_{float(i)}']) / hl.agg.sum(mt.n)
+                    for i in [1] + list(range(5,31,5)) +  [50, 100]
+                }
+            ),
+            hl.struct(
+                mean=hl.agg.sum(hl.cond(mt.sex == 'male', mt.mean * mt.n, 0.5*mt.mean * mt.n))/hl.agg.sum(mt.n),
+                median=hl.median(hl.agg.collect(hl.cond(mt.sex == 'male', mt.median, 0.5*mt.median))),
+                **{
+                    f'over_{i/2}':
+                    hl.null(hl.tfloat32) if not f'over_{i/2}' in mt.entry else
+                    hl.agg.sum(
+                        hl.cond(
+                            mt.sex == 'male',
+                            mt[f'over_{i/2}'],
+                            mt[f'over_{float(i)}']
+                        )
+                    ) / hl.agg.sum(mt.n)
+                    for i in [1] + list(range(5,31,5)) +  [50, 100]
+                }
+            )
+        )
+
+    def get_cov_ht(data_type: str) -> hl.Table:
+        cov_mt = hl.read_matrix_table(coverage_mt_path(data_type, grouped=True))
+        qc_platforms = cov_mt.aggregate_cols(hl.agg.collect_as_set(cov_mt.qc_platform))
+        return cov_mt.select_rows(
+            **{
+                f'{data_type}_{platform}': hl.agg.filter(
+                    cov_mt.qc_platform == platform,
+                    get_haploid_coverage_struct(cov_mt)
+                ) for platform in qc_platforms
+            },
+            **{f'{data_type}_all': get_haploid_coverage_struct(cov_mt)}
+        ).rows()
+
+    genomes_cov = get_cov_ht('genomes')
+    exomes_cov = get_cov_ht('exomes')
+    ucsc_mappability = get_ucsc_mappability()
+    gtf = import_cds_from_gtf()
+
+    gtf = gtf.annotate(
+        **exomes_cov[gtf.locus],
+        **genomes_cov[gtf.locus],
+        mappability=ucsc_mappability[gtf.locus].duke_35_map
+    )
+    gtf.write('gs://gnomad-lfran/exomes_genomes_coverage/gencode_grch37.gene_by_base.cov.ht', overwrite=True)
```

Comment:
move to gnomad-resources maybe?

Reply:
also parameterize overwrite here

Reply:
This seems aggregated enough that it could go in 
`gs://gnomad-public/papers/2019-flagship-lof/v1.1/summary_gene_coverage/`
along with the tsv?

Reply:
This is just per-base, per-platform/sex coverage right (which you later aggregate over gene to get the per-gene file that is publicly released)? If so, then yes.